velox/dwio/dwrf/reader/ReaderBase.h

```diff
@@ -62,23 +62,8 @@ class ReaderBase {
  public:
   /// Creates reader base from buffered input.
   ReaderBase(
-      memory::MemoryPool& pool,
-      std::unique_ptr<dwio::common::BufferedInput> input,
-      std::shared_ptr<dwio::common::encryption::DecrypterFactory>
-          decryptorFactory = nullptr,
-      uint64_t footerEstimatedSize =
-          dwio::common::ReaderOptions::kDefaultFooterEstimatedSize,
-      uint64_t filePreloadThreshold =
-          dwio::common::ReaderOptions::kDefaultFilePreloadThreshold,
-      dwio::common::FileFormat fileFormat = dwio::common::FileFormat::DWRF,
-      bool fileColumnNamesReadAsLowerCase = false,
-      std::shared_ptr<random::RandomSkipTracker> randomSkip = nullptr,
-      std::shared_ptr<velox::common::ScanSpec> scanSpec = nullptr);
-
-  ReaderBase(
-      memory::MemoryPool& pool,
-      std::unique_ptr<dwio::common::BufferedInput> input,
-      dwio::common::FileFormat fileFormat);
+      const dwio::common::ReaderOptions& options,
```

Comment:
@wypb We also need another one for `ReaderBase(memory::MemoryPool&, std::unique_ptr<dwio::common::BufferedInput>)` (and possibly the default ones, but let's create this one first)

Reply:
Hi @Yuhta I have rolled back this part of the code. Please help me review it, thank you.

Reply:
We are missing one for 

```c++
ReaderBase(
      memory::MemoryPool& pool,
      std::unique_ptr<dwio::common::BufferedInput> input)
```

Could be just adding a default argument for `fileFormat` in the signature below.

Reply:
Hi @Yuhta  Sorry, I misunderstood, I have already given a default value to the `fileFormat` parameter. Please help me review it, thank you.